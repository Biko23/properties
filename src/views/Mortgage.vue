<template>
<div>
    <top-nav />
    <main-nav />
    <v-container id="container" fluid>
        <v-carousel cycle hide-delimiter-background show-arrows-on-hover>
            <v-carousel-item v-for="(image,i) in images" :key="i" :src="image.src">
                <v-row class="fill-height" align="center" justify="center">
                    <div class="text-h4">
                        <p id="intro">{{image.title}}</p>
                        <p id="intro2">{{image.subTitle}}</p>
                    </div>
                </v-row>
            </v-carousel-item>
        </v-carousel>
    </v-container>

    <br />
    <div style="text-align: center; margin-top: 15px">
        <h3><span style="color: #3b6ef3"> HOW </span>IT WORKS</h3>
    </div>
    <v-container>
        <v-row>
            <v-col></v-col>
            <v-col cols="12" md="5" style="background-color: #f2f2f2">
                <div>
                    <v-card-title>Stanbic Bank</v-card-title>
                    <v-card-subtitle>
                        It can be possible to start constructing your rentals or dream home with House Construction financing today
                        <!-- <br /> -->
                        <!-- <v-row row d-flex>
                            <v-text-field v-model="name" label="Address" required outlined></v-text-field>
                            <v-btn style="
                    background-color: #3b6ef3;
                    margin-top: 0.5em;
                    margin-left: 0.5em;
                  ">
                                <span style="
                      color: #ffffff;
                      font-size: 20px;
                      font-style: normal;
                      font-weight: 300;
                      line-height: 30px;
                      letter-spacing: 0em;
                      text-align: center;
                      text-transform: capitalize;
                    ">
                                    See Offer</span>
                            </v-btn>
                        </v-row> -->
                    </v-card-subtitle>
                    <v-card-subtitle class="loan-heading">MAXIMUM LOAN AMOUNT</v-card-subtitle>
                    <v-card-subtitle class="loan-subheading">UGX 1 Billion</v-card-subtitle>
                    <v-card-title class="loan-heading">MINIMUM AMOUNT AVAILABLE</v-card-title>
                    <v-card-subtitle class="loan-subheading">UGX 30 Million</v-card-subtitle>
                    <!-- <v-card-title class="loan-heading">FLEXIBLE REPAYMENT PERIODS</v-card-title>
                    <v-card-subtitle class="loan-subheading">20 years</v-card-subtitle> -->
                </div>
            </v-col>
            <v-col cols="12" md="5">
                <v-card class="mx-auto" max-width="auto">
                    <v-img src="../assets/construction.jpg" height="250px"></v-img>
                    <v-card-title class="text-h5" style="dsplay: inline; text-align: center; color: blue;">
                        House Construction Loan
                    </v-card-title>
                </v-card>
            </v-col>
            <v-col></v-col>
        </v-row><br /><br>

        <v-row>
            <v-col></v-col>
            <v-col cols="12" md="5">
                <v-card max-width="auto">
                    <v-img src="../assets/housepurchase.jpg" height="250px"></v-img>
                    <v-card-title class="text-h5" style="dsplay: inline; text-align: center; color: blue;">
                        House Purchase Loan
                    </v-card-title>
                </v-card>
            </v-col>
            <v-col cols="12" md="5" style="background-color: #f2f2f2">
                <div>
                    <v-card-title>Stanbic Bank</v-card-title>
                    <v-card-subtitle>
                        Are you are looking to buy a house or flat? Buy an existing house, condominium or apartment with this great mortgage solution. It can be possible to be a home owner at the best rates on the market. Get your property with reputable developers, with affordable insurance from our trusted partners.

                        <!-- <p style="text-align: center; margin-top: 1rem">
                            <v-btn style="background-color: #3b6ef3">
                                <span style="
                      color: #ffffff;
                      font-size: 20px;
                      font-style: normal;
                      font-weight: 300;
                      line-height: 30px;
                      letter-spacing: 0em;
                      text-align: center;
                      text-transform: capitalize;
                    ">
                                    Get a Broker</span>
                            </v-btn>
                        </p> -->
                    </v-card-subtitle>
                    <v-card-subtitle class="loan-heading">MAXIMUM LOAN AMOUNT</v-card-subtitle>
                    <v-card-subtitle class="loan-subheading">UGX 1 Billion</v-card-subtitle>
                    <v-card-title class="loan-heading">MINIMUM AMOUNT AVAILABLE</v-card-title>
                    <v-card-subtitle class="loan-subheading">UGX 30 Million</v-card-subtitle>
                </div>
            </v-col>
            <v-col></v-col>
        </v-row><br><br>

        <v-row><br />
            <v-col></v-col>
            <v-col cols="12" md="5" style="background-color: #f2f2f2">
                <div>
                    <v-card-title>Stanbic Bank</v-card-title>

                    <v-card-subtitle>
                        Do you need to complete your dream home? It Can Be possible to meet your goal.
                        <!-- <p style="text-align: center; margin-top: 1rem">
                            <v-btn style="background-color: #3b6ef3">
                                <span style="
                      color: #ffffff;
                      font-size: 20px;
                      font-style: normal;
                      font-weight: 300;
                      line-height: 30px;
                      letter-spacing: 0em;
                      text-align: center;
                      text-transform: capitalize;
                    ">
                                    Sell
                                </span>
                            </v-btn>
                        </p> -->
                    </v-card-subtitle>
                    <v-card-subtitle class="loan-heading">MAXIMUM LOAN AMOUNT</v-card-subtitle>
                    <v-card-subtitle class="loan-subheading">UGX 1 Billion</v-card-subtitle>
                    <v-card-title class="loan-heading">MINIMUM AMOUNT AVAILABLE</v-card-title>
                    <v-card-subtitle class="loan-subheading">UGX 30 Million</v-card-subtitle>
                </div>
            </v-col>
            <v-col cols="12" md="5">
                <v-card max-width="auto">
                    <v-img src="../assets/finish2.jpg" height="200px"></v-img>
                    <v-card-title class="text-h5" style="dsplay: inline; text-align: center; color: blue;">
                        House Completion Loan
                    </v-card-title>
                </v-card>
            </v-col>
            <v-col></v-col>
        </v-row>
    </v-container>
    <Footer />
</div>
</template>

<script>
import Footer from "../components/Footer";
import MainNav from '../components/MainNav.vue';
import TopNav from '../components/TopNav.vue';
export default {
    components: {
        Footer,
        TopNav,
        MainNav
    },
    name: "Mortgage",
    data() {
        return {
            images: [{
                    src: 'https://res.cloudinary.com/diued7ugb/image/upload/v1626688778/dillon-kydd-XGvwt544g8k-unsplash_lrvzid.jpg',
                    title: 'Financing your next home starts here.',
                    subTitle: 'Directly to us, through an agent, or yourself'
                },
                {
                    src: 'https://res.cloudinary.com/diued7ugb/image/upload/v1626688591/vj-von-art-BP1Ze0qcp-c-unsplash_tlgt7k.jpg',
                    title: 'Financing your next home starts here.',
                    subTitle: 'Directly to us, through an agent, or yourself'
                },
                {
                    src: 'https://res.cloudinary.com/diued7ugb/image/upload/v1626688778/dillon-kydd-XGvwt544g8k-unsplash_lrvzid.jpg',
                    title: 'Financing your next home starts here.',
                    subTitle: 'Directly to us, through an agent, or yourself'
                }
            ]
        }
    }
};
</script>

<style scoped>
#intro {
    color: #ffffff;
    font-size: 2rem;
    font-weight: 400;
    text-align: center;
    margin-top: 180px;
}

#intro2 {
    color: #ffffff;
    font-size: 1.2rem;
    font-weight: 400;
    text-align: center;
}

.loan-heading {
    font-size: 16px;
    font-weight: 500;
    margin-top: 0;
    margin-bottom: 0;
    padding-top: 0;
    padding-bottom: 0;
}

.loan-subheading {
    margin-top: 0;
    margin-bottom: 0;
    padding-top: 0;
    font-size: 16px;
}

#card {
    background-color: #f4f5f8;
}

#container {
    padding: 0;
    margin: 0 0 1rem 0;
    height: 75vh;
    width: 100%;
}

@media only screen and (max-width: 768px) {
    #container {
        height: auto;
        width: auto;
    }

    #intro {
        color: #ffffff;
        font-size: 1rem;
        font-weight: 400;
        text-align: center;
        margin-top: auto;
    }

    #intro2 {
        color: #ffffff;
        font-size: 0.8rem;
        font-weight: 400;
        text-align: center;
    }

    #col-hiden {
        display: none;
    }
}
</style>
