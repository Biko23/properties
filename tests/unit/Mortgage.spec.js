import { createLocalVue, mount } from '@vue/test-utils'
import Mortgage from '@/views/Mortgage.vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'


describe('Mortgage.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let actions
    let getters
    let store
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            postAUserLog: jest.fn()           
        }
        store = new Vuex.Store({
            actions
        })
    })

    const mountFactory = options => {
        return mount(Mortgage, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFactory({})
        expect(wrapper.html()).toMatchSnapshot()
    })
})

