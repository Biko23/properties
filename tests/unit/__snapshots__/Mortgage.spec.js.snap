// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mortgage.vue should have a match snapshot 1`] = `
<div>
  <div class="container container--fluid" id="container">
    <div class="v-window v-item-group theme--dark v-window--show-arrows-on-hover v-carousel v-carousel--hide-delimiter-background" style="height: 500px;">
      <div class="v-window__container">
        <transition-stub name="">
          <div class="v-window-item">
            <div class="v-image v-responsive v-carousel__item theme--light" style="height: 500px;">
              <transition-stub name="fade-transition" mode="in-out">
                <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
              </transition-stub>
              <div class="v-responsive__content">
                <div class="row fill-height align-center justify-center">
                  <div class="text-h4">
                    <p id="intro">Financing your next home starts here.</p>
                    <p id="intro2">Directly to us, through an agent, or yourself</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </transition-stub>
        <transition-stub name="">
          <!---->
        </transition-stub>
        <transition-stub name="">
          <!---->
        </transition-stub>
        <div class="v-window__prev"><button type="button" class="v-btn v-btn--icon v-btn--round theme--dark v-size--default" aria-label="Previous visual"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--dark" style="font-size: 36px;"></i></span></button></div>
        <div class="v-window__next"><button type="button" class="v-btn v-btn--icon v-btn--round theme--dark v-size--default" aria-label="Next visual"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--dark" style="font-size: 36px;"></i></span></button></div>
      </div>
      <div class="v-carousel__controls">
        <div class="v-item-group theme--dark"></div>
      </div>
    </div>
  </div> <br>
  <div style="text-align: center; margin-top: 15px;">
    <h3><span style="color: rgb(59, 110, 243);"> HOW </span>IT WORKS</h3>
  </div>
  <div class="container">
    <div class="row">
      <div class="col"></div>
      <div class="col-md-5 col-12" style="background-color: rgb(242, 242, 242);">
        <div>
          <div class="v-card__title">Stanbic Bank</div>
          <div class="v-card__subtitle">
            It can be possible to start constructing your rentals or dream home with House Construction financing today
          </div>
          <div class="v-card__subtitle loan-heading">MAXIMUM LOAN AMOUNT</div>
          <div class="v-card__subtitle loan-subheading">UGX 1 Billion</div>
          <div class="v-card__title loan-heading">MINIMUM AMOUNT AVAILABLE</div>
          <div class="v-card__subtitle loan-subheading">UGX 30 Million</div>
        </div>
      </div>
      <div class="col-md-5 col-12">
        <div class="mx-auto v-card v-sheet theme--light" style="max-width: auto;">
          <div class="v-image v-responsive theme--light" style="height: 250px;">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
          <div class="v-card__title text-h5" style="text-align: center; color: blue;">
            House Construction Loan
          </div>
        </div>
      </div>
      <div class="col"></div>
    </div><br><br>
    <div class="row">
      <div class="col"></div>
      <div class="col-md-5 col-12">
        <div class="v-card v-sheet theme--light" style="max-width: auto;">
          <div class="v-image v-responsive theme--light" style="height: 250px;">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
          <div class="v-card__title text-h5" style="text-align: center; color: blue;">
            House Purchase Loan
          </div>
        </div>
      </div>
      <div class="col-md-5 col-12" style="background-color: rgb(242, 242, 242);">
        <div>
          <div class="v-card__title">Stanbic Bank</div>
          <div class="v-card__subtitle">
            Are you are looking to buy a house or flat? Buy an existing house, condominium or apartment with this great mortgage solution. It can be possible to be a home owner at the best rates on the market. Get your property with reputable developers, with affordable insurance from our trusted partners.

          </div>
          <div class="v-card__subtitle loan-heading">MAXIMUM LOAN AMOUNT</div>
          <div class="v-card__subtitle loan-subheading">UGX 1 Billion</div>
          <div class="v-card__title loan-heading">MINIMUM AMOUNT AVAILABLE</div>
          <div class="v-card__subtitle loan-subheading">UGX 30 Million</div>
        </div>
      </div>
      <div class="col"></div>
    </div><br><br>
    <div class="row"><br>
      <div class="col"></div>
      <div class="col-md-5 col-12" style="background-color: rgb(242, 242, 242);">
        <div>
          <div class="v-card__title">Stanbic Bank</div>
          <div class="v-card__subtitle">
            Do you need to complete your dream home? It Can Be possible to meet your goal.
          </div>
          <div class="v-card__subtitle loan-heading">MAXIMUM LOAN AMOUNT</div>
          <div class="v-card__subtitle loan-subheading">UGX 1 Billion</div>
          <div class="v-card__title loan-heading">MINIMUM AMOUNT AVAILABLE</div>
          <div class="v-card__subtitle loan-subheading">UGX 30 Million</div>
        </div>
      </div>
      <div class="col-md-5 col-12">
        <div class="v-card v-sheet theme--light" style="max-width: auto;">
          <div class="v-image v-responsive theme--light" style="height: 250px;">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
          <div class="v-card__title text-h5" style="text-align: center; color: blue;">
            House Completion Loan
          </div>
        </div>
      </div>
      <div class="col"></div>
    </div>
  </div>
</div>
`;
